### ISSUE TITLE
Collection element subtyping rule applied to ".class" expressions and instanceof operator

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#84]
The rule that the element type of a collection must be a subtype of the collection itself is applied to ".class" expressions, even though this doesn't affect behavior. Consider this class

```java
import org.checkerframework.checker.determinism.qual.*;
import java.util.*;

public class CollectionTypes {

    public static void f() {
        Class<?> c = Collection.class;
    }
}
```
```
CollectionTypes.java:7: error: [invalid.upper.bound.on.type.argument] upper bound of type argument (@NonDet) must be a subtype of the collection type (@Det)
        Class<?> c = Collection.class;
                               ^
```

An expression like this represents a Java class, and doesn't affect behavior, so there shouldn't be an error. Also, there's no way to get around this error, it must be suppressed.

We get the same error with the `instanceof` operator.

```java
        boolean b = (c instanceof Collection);
```
```
CollectionTypes.java:7: error: [invalid.upper.bound.on.type.argument] upper bound of type argument (@NonDet) must be a subtype of the collection type (@Det)
        boolean b = (c instanceof Collection);
                                  ^
```
Similarly, this shouldn't be an error because `Collection` is not being used to instantiate any real Object.

**Comment by smillst** [comment]
Fixed in 061ffb8.

**Comment by jwaataja** [comment]
The original issue with the `instanceof` operator still remains. Sorry it wasn't in the original title, I've update it.

**Comment by smillst** [comment]
Apologies for not reading more carefully.   I fixed the `instanceOf` operator in 296b191.
